### ISSUE TITLE
[Feature Request] Add support for identifier renaming and find all references

### ISSUE THREAD

**Issue by soraros** [modular/mojo#1233]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

As title. Go to definition does work now, and I think that already laid the ground work for renaming and find all usage.

### What is your motivation for this change?

Better user experience for VSCode extension.

### Any other details?

NA

**Comment by Prodesire** [comment]
Agreed! I really hope that the vscode mojo plugin can provide refactoring capabilities.

**Comment by River707** [comment]
Thanks for filing this! These are definitely on the roadmap to add, we'll use this issue to track when those get covered.

**Comment by walter-erquinigo** [comment]
As a note, find all references has been implemented a few months ago with the limitation of not showing cross-file references. This will be improved later this year, as well as renaming support, which is still in the roadmap.

**Comment by tristanbiesecker** [comment]
+1. It would be really nice to have these capabilities supported when working with Mojo.
